dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java

```diff
@@ -67,12 +73,41 @@ public Experiment save(final Experiment experiment, final User user) throws
 
         builder.modDate(Instant.now());
         builder.lastModifiedBy(user.getUserId());
+        
+        if(experiment.goals().isPresent()) {
+            validateGoals(experiment.goals().get());
+        }
 
         final Experiment experimentToSave = builder.build();
 
         return factory.save(experimentToSave);
     }
 
+    private void validateGoals(Goals goals) {
```

Comment:
I don't know if this method should be here, really it is about Metrics not goals so I don't know if we should create something like MetrictUtil, maybe we are going to need to reuse this method in the future when Metricts start working out of Experiments

Reply:
Done. Moved to MetricsUtil.